### PR TITLE
If enabled, print log framing content to stdout/stderr

### PR DESCRIPTION
**Pull request by harryjsmith**
Helps stdout/stderr parsers associate a request ID with a sequence of logs

### CODE REVIEW

File: lib/fdk/listener.rb [hunk 1]
```diff
@@ -46,23 +51,41 @@ def listen(&block)
...
     def private_socket_path
       socket_path + ".private"
     end
+
+    def log_frame_header(headers)
+      return if @fn_logframe_name.nil? || @fn_logframe_hdr.nil?
+
+      k = @fn_logframe_hdr.downcase
+      v = headers[k]
+      return if v.nil?
```
Comment (rdallman) [inline lib/fdk/listener.rb]: maybe `.empty?` instead of nil here, since it's possibly an empty string?

Comment (rdallman) [reply lib/fdk/listener.rb]: empty list or empty string I suppose

File: lib/fdk/listener.rb [hunk 2]
```diff
@@ -46,23 +51,41 @@ def listen(&block)
...
     def socket_path
       @socket_path ||= url[5..url.length]
     end
 
     def private_socket_path
       socket_path + ".private"
     end
+
+    def log_frame_header(headers)
+      return if @fn_logframe_name.nil? || @fn_logframe_hdr.nil?
```
Comment (rdallman) [inline lib/fdk/listener.rb]: maybe same thing here, re: empty (empty string)

File: lib/fdk/listener.rb (hunk 1 above)
Comment (harryjsmith) [reply lib/fdk/listener.rb]: Fixed. Have to do a nil check first anyway unfortunately else you get `Error in request handling undefined method `empty?' for nil:NilClass`

File: lib/fdk/listener.rb (hunk 2 above)
Comment (harryjsmith) [reply lib/fdk/listener.rb]: Done. Had to refactor it out into a separate method to keep rubocop happy (else `Metrics/CyclomaticComplexity: Cyclomatic complexity for log_frame_header is too high`)

**Comment by crush-157** [comment]
@harryjsmith Can you bump the Gem version to 20 please?  Otherwise looks OK.

**Comment by crush-157** [comment]
That's in version.rb and in the examples Gemfile

**Comment by harryjsmith** [comment]
@crush-157 Done, thanks

**Pull request merged**